File: miraculous-frontend/miraculous-ui/src/dto/track-link-dto/track-link-dto.ts
export class TrackLinkDto {

  id: number;
  title: string;
  artistId: number;
  artistLogin: string;
  albumId: number;
  albumTitle: string;
  duration: number;
  numberOfLikes: number;
  coverPictureUrlName: string;
  isFree: boolean;
  genres: string[];
  price: number;
  inCartOfCurrentVisitor: boolean;


  constructor(id?: number, title?: string, artistId?: number, artistLogin?: string, albumId?: number, albumTitle?: string,
              duration?: number, numberOfLikes?: number, coverPictureUrl?: string, isFree?: boolean, genres?: string[],
              price?: number, inCartOfCurrentVisitor? :boolean) {
    this.id = id;
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.albumId = albumId;
    this.albumTitle = albumTitle;
    this.duration = duration;
    this.numberOfLikes = numberOfLikes;
    this.coverPictureUrlName = coverPictureUrl;
    this.isFree = isFree;
    this.genres = genres;
    this.price = price;
    this.inCartOfCurrentVisitor = inCartOfCurrentVisitor;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/TrackLinkDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TrackLinkDto {

    private Long id;
    private String title;
    private Long artistId;
    private String artistLogin;
    private Long albumId;
    private String albumTitle;
    private Double duration;
    private Long numberOfLikes;
    private String coverPictureUrlName;
    private Boolean isFree;
    private List<String> genres;
    private Double price;
    private boolean isLikedByCurrentVisitor;
    private boolean isInCartOfCurrentVisitor;
}
